mcp/server.go

```diff
@@ -46,6 +46,14 @@ type Server struct {
 	resourceSubscriptions   map[string]map[*ServerSession]bool // uri -> session -> bool
 }
 
+// InitialCapabilities allows a server to advertise support for capabilities
+// during initialization, even if no resources for that capability have been added yet.
+type InitialCapabilities struct {
```

Comment:
I don't think we need a new type for these. Just add them to ServerOptions.
Let's call them `HasTools`, etc. unless we can think of a better name.

Reply:
Sounds good, done.